apps/server/src/app/fhir.controller.ts

```diff
@@ -95,11 +96,83 @@ export class FhirController extends BaseController {
       method: 'DELETE'
     };
 
-    this.logger.log('Sending PUT request to FHIR server with the new resource ID');
-
     // Create the new resource with the new id
     await this.httpService.request(createOptions).toPromise();
 
+    const searchForReference = (searchResourceType: string, searchParameter: string) => {
+
+      return new Promise(async (resolve) => {
+        const params = {};    // DONT use _summary=true. The results may end up getting used to update the resource.
+        params[searchParameter] = `${resourceType}/${currentId}`;
+        const searchUrl = buildUrl(fhirServerBase, searchResourceType, null, null, params);
+
+        const results = await this.httpService.get(searchUrl).toPromise();
+        const bundle: Bundle = results.data;
+        const resources = (bundle.entry || []).map(entry => <DomainResource> entry.resource);
+
+        resolve(resources);
+      });
+    };
+
+    const searchPromises = [];
+    // These search parameters apply to both STU3 and R4 servers
+    searchPromises.push(searchForReference('ImplementationGuide', 'resource'));
```

Comment:
please also search the "global" parameter.
note that this will also need to be updated in the unit test with 
`.get('/ImplementationGuide?global=StructureDefinition%2Ftest')`